components/edit_post/edit_post.tsx

```diff
@@ -0,0 +1,481 @@
+    prevShowState: boolean;
+};
+
+const {KeyCodes} = Constants;
+
+const TOP_OFFSET = 0;
+const RIGHT_OFFSET = 10;
+
+const EditPost = ({editingPost, actions, ...rest}: Props): JSX.Element => {
+    const [editText, setEditText] = useState<string>(
+        editingPost.post?.message_source || editingPost.post?.message || '',
+    );
+    const [caretPosition, setCaretPosition] = useState<number>(0);
+    const [postError, setPostError] = useState<React.ReactNode | null>(null);
+    const [errorClass, setErrorClass] = useState<string>('');
+    const [showEmojiPicker, setShowEmojiPicker] = useState<boolean>(false);
+    const [renderScrollbar, setRenderScrollbar] = useState<boolean>(false);
+
+    const textboxRef = useRef<TextboxClass>(null);
+    const emojiButtonRef = useRef<null>(null);
+
+    const {formatMessage} = useIntl();
+
+    const scrollOptions: boolean|ScrollIntoViewOptions = isInternetExplorer() || isSafari() ? false : {behavior: 'smooth', block: 'end'};
+
+    useEffect(() => textboxRef?.current?.focus(), []);
+    useEffect(() => {
+        const parentId = `${editingPost.isRHS ? 'rhsP' : 'p'}ost_${editingPost.postId}`;
+        const parentElement = document.getElementById(parentId);
+
+        parentElement?.scrollIntoView?.(scrollOptions);
+    }, []);
+    useEffect(() => {
+        const handlePaste = (e: ClipboardEvent) => {
+            if (
+                !e.clipboardData ||
+                !e.clipboardData.items ||
+                !rest.canEditPost ||
+                (e.target as HTMLTextAreaElement).id !== 'edit_textbox'
+            ) {
+                return;
+            }
+
+            const {clipboardData} = e;
+            const table = getTable(clipboardData);
+
+            if (!table) {
+                return;
+            }
+
+            e.preventDefault();
+
+            let message = editText;
+            let newCaretPosition = caretPosition;
+
+            if (table && isGitHubCodeBlock(table.className)) {
+                const {formattedMessage, formattedCodeBlock} = formatGithubCodePaste(
+                    caretPosition,
+                    message,
+                    clipboardData,
+                );
+                newCaretPosition = caretPosition + formattedCodeBlock.length;
+                message = formattedMessage;
+            } else if (table) {
+                message = formatMarkdownTableMessage(table, editText.trim(), newCaretPosition);
+                newCaretPosition = message.length - (editText.length - newCaretPosition);
+            }
+
+            setEditText(message);
+            setCaretPosition(newCaretPosition);
+
+            if (textboxRef.current) {
+                Utils.setCaretPosition(textboxRef.current.getInputBox(), newCaretPosition);
+            }
+        };
+
+        document.addEventListener('paste', handlePaste);
+
+        return () => {
+            document.removeEventListener('paste', handlePaste);
+        };
+    });
```

Comment:
Do we want to run this on every render?

Reply:
Probably not worth it, but have you consider make the handlePaste a callback, and make the effect only about adding and removing the event (and depending just on the handler)?

Reply:
nice catch ... I forgot to attach the `[]` ... thanks.

Unfortunately you have to define these inside the `useEffect` to be effective ... :/